onair/src/reasoning/agent.py

```diff
@@ -13,29 +13,40 @@
 """
 from ..ai_components.learners_interface import LearnersInterface
 from ..ai_components.planners_interface import PlannersInterface
+from ..reasoning.complex_reasoning_interface import ComplexReasoningInterface
 from ..reasoning.diagnosis import Diagnosis
 
 class Agent:
-    def __init__(self, vehicle, plugin_list):
+    def __init__(self, vehicle, learners_plugin_dict, planners_plugin_dict, complex_plugin_dict):
+
         self.vehicle_rep = vehicle
         self.mission_status = self.vehicle_rep.get_status()
         self.bayesian_status = self.vehicle_rep.get_bayesian_status()
 
         # AI Interfaces
-        self.learning_systems = LearnersInterface(self.vehicle_rep.get_headers(),plugin_list)
-        self.planning_systems = PlannersInterface(self.vehicle_rep.get_headers(),plugin_list)
+        self.learning_systems = LearnersInterface(self.vehicle_rep.get_headers(),learners_plugin_dict)
+        self.planning_systems = PlannersInterface(self.vehicle_rep.get_headers(),planners_plugin_dict)
+        self.complex_reasoning_systems = ComplexReasoningInterface(self.vehicle_rep.get_headers(),complex_plugin_dict)
+
+      
+    def render_reasoning(self):
+        return self.complex_reasoning_systems.render_reasoning()
 
-    # Markov Assumption holds 
     def reason(self, frame):
```

Comment:
Did we rename "frame" to "low level data" elsewhere?

Reply:
The rename should happen everywhere

Reply:
We will make this change later, there is some difficulty doing it quickly because frame, frames, updated_frame -> low_level_data, low_level_datas, update_low_level_data. I will add a ticket for the change after this is merged.